pfcli/service/client/checkpoint.py

```diff
@@ -198,6 +195,9 @@ def upload_files(
         mpu_url_dicts: List[Dict[str, Any]],
         source_path: Path,
         expand: bool,
+        max_workers: int = min(
+            32, (os.cpu_count() or 1) + 4
+        ),  # default of ``ThreadPoolExecutor``
```

Comment:
Why `os.cpu_count() or 1` instead of just `os.cpu_count()`?
Could `os.cpu_count()` return `None`?

Reply:
The default value of `max_workers` comes from the default value for `ThreadPoolExecutor`.

```
        if max_workers is None:
            # ThreadPoolExecutor is often used to:
            # * CPU bound task which releases GIL
            # * I/O bound task (which releases GIL, of course)
            #
            # We use cpu_count + 4 for both types of tasks.
            # But we limit it to 32 to avoid consuming surprisingly large resource
            # on many core machine.
```